### PR TITLE
Revertchanges

### PR DESCRIPTION
**Pull request by KalyanBoya**
What does this PR do?
Updating the reference implementation to have 'withsharing' keyword

What issues does this PR fix or reference?
@W-13177204@
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001R3fDoYAJ/view

Functionality Before
remove record level   access checks

Functionality After
This would not enforce record level access checks for checkout integrations.

How to Test/Testing Effort
run the checkout

### CODE REVIEW

File: examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls [hunk 1]
```diff
@@ -258,7 +259,7 @@ global class B2CDeliverySample implements sfdc_checkout.CartShippingCharges {
         for (ShippingOptionsAndRatesFromExternalService shippingOption: shippingOptions) {
             String shippingOptionNumber = String.valueOf(i);
             String name = defaultDeliveryMethodName + shippingOptionNumber;
-            List<OrderDeliveryMethod> odms = [SELECT Id, ProductId, Carrier, ClassOfService FROM OrderDeliveryMethod WHERE Name = :name];
+            List<OrderDeliveryMethod> odms = [SELECT Id, ProductId, Carrier, ClassOfService FROM OrderDeliveryMethod WHERE Name = :name WITH SECURITY_ENFORCED];
```
Comment (snilakandan13) [inline examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:262]: Do we want to introduce `WITH SECURITY_ENFORCED`? I did not see that in Aditya's PR

Comment (snilakandan13) [reply examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:262]: These are his changes - https://github.com/forcedotcom/commerce-on-lightning/pull/214/files

Comment (snilakandan13) [reply examples/b2c/checkout/integrations/classes/B2CDeliverySample.cls:262]: It looks like with this, there are no ODMs created

**Review by KalyanBoya — COMMENTED** [review]
@snilakandan13 @mkosak-sf please review

**Review by snilakandan13 — APPROVED** [review]

**Review by unandyala — APPROVED** [review]

**Review by gkhunger-sf — APPROVED** [review]

**Pull request merged**